pkg/component/spec.go

```diff
@@ -81,9 +81,22 @@ func (t *CommandTimeoutSpec) InitDefaults() {
 	t.Stop = 30 * time.Second
 }
 
+// ServiceTimeoutSpec is the timeout specification for subprocess.
+type ServiceTimeoutSpec struct {
+	Checkin time.Duration `config:"checkin" yaml:"checkin"`
+}
+
+// InitDefaults initialized the defaults for the timeouts.
+func (t *ServiceTimeoutSpec) InitDefaults() {
+	t.Checkin = 30 * time.Second
+}
+
 // ServiceSpec is the specification for an input that executes as a service.
 type ServiceSpec struct {
+	Name       string                `config:"name" yaml:"name" validate:"required"`
+	Service    int                   `config:"service" yaml:"service" validate:"required"`
```

Comment:
I think calling this `service` already nested under another `service` does not really explain what this is for. A more descriptive name would be better.

How about `creds_port`?

Reply:
Agreed. However, I'd suggest to use a full descriptive name, not an abbreviation. Why not call it `port`?

Reply:
It was called "service" in the previous V1 implementation, was not sure what's the better name. The "port" is too generic, this is specific "hack" for the creds exchange with the service, "cport" it is then